pkg/sdk/security_integrations_def.go

```diff
@@ -0,0 +1,231 @@
+	OptionalBooleanAssignment("SYNC_PASSWORD", g.ParameterOptions()).
+	OptionalComment().
+	WithValidation(g.AtLeastOneValueSet, "Enabled", "NetworkPolicy", "SyncPassword", "Comment")
+
+var scimIntegrationUnsetDef = g.NewQueryStruct("SCIMIntegrationUnset").
+	OptionalSQL("ENABLED").
+	OptionalSQL("NETWORK_POLICY").
+	OptionalSQL("SYNC_PASSWORD").
+	OptionalSQL("COMMENT").
+	WithValidation(g.AtLeastOneValueSet, "Enabled", "NetworkPolicy", "SyncPassword", "Comment")
+
+var SecurityIntegrationsDef = g.NewInterface(
+	"SecurityIntegrations",
+	"SecurityIntegration",
+	g.KindOfT[AccountObjectIdentifier](),
+).
+	CustomOperation(
+		"CreateSAML2",
+		"https://docs.snowflake.com/en/sql-reference/sql/create-security-integration-saml2",
+		createSecurityIntegrationOperation("CreateSAML2Integration", func(qs *g.QueryStruct) *g.QueryStruct {
+			return qs.
+				PredefinedQueryStructField("integrationType", "string", g.StaticOptions().SQL("TYPE = SAML2")).
+				BooleanAssignment("ENABLED", g.ParameterOptions().Required()).
+				TextAssignment("SAML2_ISSUER", g.ParameterOptions().Required().SingleQuotes()).
+				TextAssignment("SAML2_SSO_URL", g.ParameterOptions().Required().SingleQuotes()).
+				TextAssignment("SAML2_PROVIDER", g.ParameterOptions().Required().SingleQuotes()).
+				TextAssignment("SAML2_X509_CERT", g.ParameterOptions().Required().SingleQuotes()).
+				ListAssignment("ALLOWED_USER_DOMAINS", "UserDomain", g.ParameterOptions().Parentheses()).
+				ListAssignment("ALLOWED_EMAIL_PATTERNS", "EmailPattern", g.ParameterOptions().Parentheses()).
+				OptionalTextAssignment("SAML2_SP_INITIATED_LOGIN_PAGE_LABEL", g.ParameterOptions().SingleQuotes()).
+				OptionalBooleanAssignment("SAML2_ENABLE_SP_INITIATED", g.ParameterOptions()).
+				OptionalTextAssignment("SAML2_SNOWFLAKE_X509_CERT", g.ParameterOptions().SingleQuotes()).
+				OptionalBooleanAssignment("SAML2_SIGN_REQUEST", g.ParameterOptions()).
+				OptionalTextAssignment("SAML2_REQUESTED_NAMEID_FORMAT", g.ParameterOptions().SingleQuotes()).
+				OptionalTextAssignment("SAML2_POST_LOGOUT_REDIRECT_URL", g.ParameterOptions().SingleQuotes()).
+				OptionalBooleanAssignment("SAML2_FORCE_AUTHN", g.ParameterOptions()).
+				OptionalTextAssignment("SAML2_SNOWFLAKE_ISSUER_URL", g.ParameterOptions().SingleQuotes()).
+				OptionalTextAssignment("SAML2_SNOWFLAKE_ACS_URL", g.ParameterOptions().SingleQuotes())
+		}),
+		userDomainDef,
+		emailPatternDef,
+	).
+	CustomOperation(
+		"CreateSCIM",
+		"https://docs.snowflake.com/en/sql-reference/sql/create-security-integration-scim",
+		createSecurityIntegrationOperation("CreateSCIMIntegration", func(qs *g.QueryStruct) *g.QueryStruct {
+			return qs.
+				PredefinedQueryStructField("integrationType", "string", g.StaticOptions().SQL("TYPE = SCIM")).
+				BooleanAssignment("ENABLED", g.ParameterOptions().Required()).
+				OptionalAssignment(
```

Comment:
Why optionalAssignment and not assignment? (I know it's not much of a difference in this case, but this is required so it would be more explicit)